src/OAuth2/Saml/Saml2Assertion.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+
+namespace OAuth2\Saml;
+
+use OneLogin_Saml2_Response;
+
+
+/**
+ * This is an adapter that uses OneLogin utils
+ * Class SamlAssertionDecoder
+ * @package OAuth2\Saml
+ */
+class Saml2Assertion implements Saml2AssertionInterface
+{
+
+    /**
+     * @var OneLogin_Saml2_Response
+     */
+    protected $saml_response;
+
+    protected $saml_settings;
+
+    protected $raw_assertion;
+
+    protected $saml_response_error;
+
+    function __construct($settings)
+    {
```

Comment:
I see a lot of extra whitespace here... can we get rid of all superfluous whitespace? I suggest running `php-cs-fixer` for this.